src/Environment.php

```diff
@@ -133,6 +130,11 @@ public function __construct(LoaderInterface $loader, $options = [])
 
         $this->addExtension(new CoreExtension());
         $this->addExtension(new EscaperExtension($options['autoescape']));
+        if (\PHP_VERSION_ID >= 80000) {
+            $this->addExtension(new YieldNotReadyExtension($this->useYield));
+        } elseif ($this->useYield) {
+            throw new \LogicException('The "use_yield" option of Twig can only be enabled when running on PHP 8.0 or later.');
```

Comment:
I suggest updating the phpdoc to describe this requirement when documenting the option.

Reply:
I removed the PHP 8 requirement: it exists just because attributes can be read only on PHP 8, but yield works will on PHP 7, so it's fine to skip the check. We can expect ppl that write new code for PHP 7 to also run it on PHP 8 so their code will be checked anyway.